csvy/validators/table_schema.py

```diff
@@ -0,0 +1,454 @@
+    """Default column validator for anything that does not have a specific one.
+
+    Attributes:
+        type_: A string specifying the type, valid values are any of the TypeEnum that
+            do not have a specific validator.
+        format_: A string specifying a format, with valid values being 'default' and
+            None.
+
+    """
+
+    type_: Literal[
+        TypeEnum.OBJECT,
+        TypeEnum.ARRAY,
+        TypeEnum.YEAR,
+        TypeEnum.YEARMONTH,
+        TypeEnum.DURATION,
+        TypeEnum.ANY,
+        None,
+    ] = Field(None, alias="type", description="A string specifying the type.")
+    format_: FormatDefault | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+class StringColumnValidator(CommonValidator):
+    """Validator for the string columns in the Table Schema.
+
+    This class is used to validate the string columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'string'.
+        format_: A string specifying a format, with valid values being 'default',
+            'email', 'uri', 'binary', 'uuid' and None.
+
+    """
+
+    type_: Literal[TypeEnum.STRING] = Field(
+        TypeEnum.STRING, alias="type", description="A string specifying the type."
+    )
+    format_: FormatString | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+class NumberColumnValidator(CommonValidator):
+    """Validator for the number columns in the Table Schema.
+
+    This class is used to validate the number columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'number'.
+        format_: A string specifying a format, with valid values being 'default' and
+            None.
+        decimalChar: The character used to separate the integer and fractional. If
+            None, '.' is used.
+        groupChar: The character used to separate groups of thousands. None assumed.
+        bareNumber: If True, the number is a bare number, without any formatting. If
+            None, True is used. If false the contents of this field may contain leading
+            and/or trailing non-numeric characters that must be stripped to parse the
+            number, eg. '95%', '£50.5'.
+
+    """
+
+    type_: Literal[TypeEnum.NUMBER] = Field(
+        TypeEnum.NUMBER, alias="type", description="A string specifying the type."
+    )
+    format_: FormatDefault | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+    decimalChar: str | None = Field(
+        None,
+        description="The character used to separate the integer and fractional. "
+        + "If None, '.' is used.",
+    )
```

Comment:
This description claims that "." is the default, but I do not see where that is set. It looks like the default is set to `None` to me.

This same pattern/comment combination is repeated multiple times in other validators. So should be checked.

Reply:
Yeah, I know. It is tricky. The point is that if this value is not provided, the default should be used - that's why is the default! - but if I dump the model again to save in the CSV file, I do not want all of those default values to be there because it will make the schema huge. And they are the defaults of the Table Schema specification, so anyone implementing the schema should know them.

In summary, what I want is that if I read the csvy file and then save it again without changes, the resulting schema declared in the header is the same, without extra information.

Any suggestion on how to tackle this better is most welcomed!

Reply:
Hmm, I guess that means the default should be used in the validation but not saved to the object that is printed as the schema. I'm not sure how the schema is being used later on (i.e for printing) though, so I can't really say much more than that.

Reply:
Let's open an issue about this and leave it for now as it is.